packages/@lwc/ssr-compiler/src/compile-js/generate-markup.ts

```diff
@@ -28,7 +28,13 @@ type RenderCallExpression = SimpleCallExpression & {
 
 const bGenerateMarkup = esTemplate`
     export async function* generateMarkup(tagName, props, attrs, slotted) {
-        attrs = attrs ?? {};
+        attrs = attrs ?? Object.create(null);
+        props = props ?? Object.create(null);
+        props = __filterProperties(
+            props,
+            new Set(${/*public fields*/ is.arrayExpression}),
+            new Set(${/*private fields*/ is.arrayExpression}),
```

Comment:
We can move the `new Set()` into the helper function to make this a bit cleaner.